src/runtime/lpython_builtin.py

```diff
@@ -481,7 +481,7 @@ def _lpython_floordiv(a: f32, b: f32) -> f32:
 
 @overload
 def _lpython_floordiv(a: i32, b: i32) -> i32:
-    r: f32
+    r: f64
     r = a/b
     result: i32
     result = int(r)
```

Comment:
Why don't we use `i64` as the type of `r`?

Reply:
It won't change the result in this case, although the `r ` variables in all overloaded functions are of floating point `f32/f64` type.

Reply:
This lead to another error being spotted - 

```
def main():
    x : f32
    y : f32
    y = 10.0
    x = 123456789.0
    print(x//y)

(lp) anurag-3:lpython user$ ./src/bin/lpython try.py && ./a.out 
12345679.000000
You have new mail in /var/mail/user
(lp) anurag-3:lpython user$ python try.py
12345678.0
```

Reply:
I don't think type of `r` matters here. `a` is `i32` and `b` is `i32` so `a/b` will anyways be `i32`. In the low level code it would just get casted to the type of `r` before storing. The final answer is already computed when `a/b` executes. Then it doesn't matter where you store it, in fact you can directly return `a/b` (conceptually, if lpython gives error then it should be fixed).

Try seeing the LLVM code being generated for the new error you found. The bug might be resolved if we carefully track what's happening at each step of code generation (AST, ASR, LLVM).

Reply:
>  The final answer is already computed when `a/b` executes. Then it doesn't matter where you store it, in fact you can directly return `a/b` (conceptually, if lpython gives error then it should be fixed).

If `r` type is irrelevent then the current implementation has loopholes because as of now you can expect different answers as per the type of r. Also directly returning `a/b` as you mentioned gives error in lpython.
> Try seeing the LLVM code being generated for the new error you found. 

Okay thanks.

Reply:
> If r type is irrelevent then the current implementation has loopholes because as of now you can expect different answers as per the type of r.

Yes, you are right, `r` type is relevant. It should be kept to the highest possible floating point kind. That way `float(a)/float(b)` won't result in rounding up and getting stored in `r`. By using anything lower than `f64` causes rounding up which leads to errors in final integer result. Suggested the same in https://github.com/lcompilers/lpython/pull/1067#discussion_r959185801